app/services/usage_monitoring/process_subscription_activity_service.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module UsageMonitoring
+  class ProcessSubscriptionActivityService < BaseService
+    Result = BaseResult
+
+    def initialize(subscription_activity:)
+      @subscription_activity = subscription_activity
+      @subscription = subscription_activity.subscription
+      super
+    end
+
+    def call
+      lifetime_usage = find_or_create_lifetime_usage
+
+      if organization.lifetime_usage_enabled? || organization.progressive_billing_enabled?
+        LifetimeUsages::CalculateService.call!(lifetime_usage:, current_usage:)
+      end
+
+      if organization.progressive_billing_enabled?
+        LifetimeUsages::CheckThresholdsService.call(lifetime_usage:)
+      end
+
+      # TODO: Add Alerting here
+
+      subscription_activity.delete
```

Comment:
You'll end up creating lots of bloat on this table, keep in mind that it will need to be `vacuumed` and also `repacked` often.

Reply:
Very good point. Who's going to do this now 😬